pandas/core/indexes/category.py

```diff
@@ -460,9 +460,7 @@ def where(self, cond, other=None):
             other = self._na_value
         values = np.where(cond, self.values, other)
 
-        cat = Categorical(values,
-                          categories=self.categories,
-                          ordered=self.ordered)
+        cat = Categorical(values, dtype=self.dtype)
```

Comment:
Hey- why can we get rid of the categories and ordered params out of interest?

Reply:
The dtype is a ``CategoricalDtype``, which contains attributes ``categories`` and ``ordered``. So we're  not really removing them, only passinf the dtype around instead, which is clearer IMO.